lab_5_scrapper/scrapper.py

```diff
@@ -140,10 +290,74 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler is a child of Crawler class.
+    """
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+        self.start_url = self.base_url
+
+        self.already_crawled_path = constants.ASSETS_PATH.parent / 'already_crawled_urls.txt'
+        with open(self.already_crawled_path, 'r', encoding='utf-8') as f:
+            self.already_crawled = set(f.readlines())
+
+        self.article_tag = 'ARTICLE URL: '
+        self.urls = [url[len(self.article_tag)+1::]
+                     for url in self.already_crawled
+                     if self.article_tag in url]
+
+    def find_articles(self) -> None:
+        response = make_request(self.start_url, self.config)
+
+        all_urls = [self.start_url]
+        if not response.ok:
+            return
+
+        article_bs = BeautifulSoup(response.text, "lxml")
+
+        all_urls.extend(self.base_url[:-len('/news'):] + link.get('href')
+                        for link in article_bs.find_all(href=True)
+                        if 'http' not in link.get('href'))
+        # all urls to my site pages are put into the html code as
+        # /something/..., but not the full link like
+        # https://www.sbras.info/news/something/...,
+        # so everything that refers to other https pages will break requests.get
+        # because it will be https://www.sbras.infohttp/...
+
+        for url in all_urls:
+            if url in self.already_crawled and url != self.base_url:
+                continue
+            elif url != self.base_url:
+                self.already_crawled.update(url)
+            self.start_url = url
+
+            counter = 0
+            extracted_url = self._extract_url(article_bs)
+            while extracted_url:
+                if self.article_tag + extracted_url in self.urls:
+                    extracted_url = self._extract_url(article_bs)
+                    continue
+
+                if len(self.urls) == self.config.get_num_articles():
+                    raise KeyboardInterrupt
+
+                self.urls.append(extracted_url)
+                self.already_crawled.update(self.article_tag + extracted_url)
+                counter += 1
+                extracted_url = self._extract_url(article_bs)
```

Comment:
how and where do I keep the list of article URLs that I am going to parse with `HTMLParser` later?

Reply:
In the Crawler.urls attribute by appending them to it.